pkg/uploader/kopia/shim.go

```diff
@@ -208,11 +208,22 @@ func (sr *shimRepository) DeleteManifest(ctx context.Context, id manifest.ID) er
 	return sr.udmRepo.DeleteManifest(ctx, udmrepo.ID(id))
 }
 
+func (sr *shimRepository) ReplaceManifests(ctx context.Context, labels map[string]string, payload interface{}) (manifest.ID, error) {
+	return manifest.ID(""), errors.New("ReplaceManifests is not supported")
+}
+
 // Flush all the unifited repository data
 func (sr *shimRepository) Flush(ctx context.Context) error {
 	return sr.udmRepo.Flush(ctx)
 }
 
+func (sr *shimRepository) ConcatenateObjects(ctx context.Context, objectIDs []object.ID) (object.ID, error) {
+	return object.ID{}, errors.New("ConcatenateObjects is not supported")
+}
+
+func (sr *shimRepository) OnSuccessfulFlush(callback repo.RepositoryWriterCallback) {
```

Comment:
Just curious about why not `OnSuccessfulFlush` function return value handling is different from other newly added functions.

Reply:
The prototype of these functions are defined by Kopia, specifically, they implements Kopia' Repository interface. In the interface, `OnSuccessfulFlush` has no return value.
On the other hand, this function is a callback under the Flush operation, while this callback is optional, if it is not to be handled, just leave it as empty.